lightning-invoice/src/payment.rs

```diff
@@ -605,23 +602,21 @@ where
 
 	// Takes in a path to have its information stored in `payment_cache`. This is done for paths
 	// that are pending retry.
-	fn process_path_inflight_htlcs(&self, payment_hash: PaymentHash, hops: Vec<RouteHop>) {
-		let payment_info_route_hops: Vec<PaymentInfoRouteHop> = hops.iter().map(|h| PaymentInfoRouteHop { pubkey: h.pubkey, short_channel_id: h.short_channel_id, fee_msat: h.fee_msat }).collect();
+	fn process_path_inflight_htlcs(&self, payment_hash: PaymentHash, hops: &Vec<RouteHop>) {
+		let payment_info_route_hops: Vec<PaymentInfoRouteHop> = hops.iter()
+			.map(|h| PaymentInfoRouteHop { pubkey: h.pubkey, short_channel_id: h.short_channel_id, fee_msat: h.fee_msat })
+			.collect();
 
 		self.payment_cache.lock().unwrap().entry(payment_hash)
-			.or_insert(PaymentInfo {
-				attempts: PaymentAttempts {
-					count: 1,
-					first_attempted_at: T::now()
-				},
-				paths: vec![]
-			})
+			.or_insert(PaymentInfo::new())
```

Comment:
Here and below, use `or_insert_with(PaymentInfo::new)` so the struct is only created when needed.